packages/repository/src/relations/has-many/has-many-through-repository.factory.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  DataObject,
+  Entity,
+  EntityCrudRepository,
+  Getter,
+  HasManyDefinition,
+} from '../..';
+import {
+  createTargetConstraint,
+  createThroughConstraint,
+  createThroughFkConstraint,
+  resolveHasManyThroughMetadata,
+} from './has-many-through.helpers';
+import {
+  DefaultHasManyThroughRepository,
+  HasManyThroughRepository,
+} from './has-many-through.repository';
+
+/**
+ * a facroty to generate hasManyThrough repository class.
+ *
+ * Warning: The hasManyThrough interface is experimental and is subject to change.
+ * If backwards-incompatible changes are made, a new major version may not be
+ * released.
+ */
+
+export type HasManyThroughRepositoryFactory<
+  TargetEntity extends Entity,
+  TargetID,
+  ThroughEntity extends Entity,
+  ForeignKeyType
+> = (
+  fkValue: ForeignKeyType,
+) => HasManyThroughRepository<TargetEntity, TargetID, ThroughEntity>;
+
+export function createHasManyThroughRepositoryFactory<
```

Comment:
Do we need a new function? Have you considered to extend the existing `createHasManyRepositoryFactory` to support both "direct" and "through" variants? Maybe that's not a good idea, I just wanted to bring it forward for consideration.

Reply:
Good point... I considered a bit. Since we only have one hasMany type, I think it might be more consistent if we only expose `createHasManyRepositoryFactory` as the entry for hasMany* relations, and let `createHasManyRepositoryFactory` to decide which factory to create.